SimplenoteTests/NSTextViewSimplenoteTests.swift

```diff
@@ -0,0 +1,250 @@
+        let range = NSRange(location: .zero, length: String.attachmentString.utf16.count)
+        textView.textStorage?.addAttribute(.attachment, value: attachment, range: range)
+
+        // Select the tail of the first line
+        let newSelectedRange = NSRange(location: textView.string.utf16.count, length: .zero)
+        textView.setSelectedRange(newSelectedRange)
+
+        XCTAssertTrue(textView.processNewlineInsertion())
+
+        // Verify there are *two* different TextAttachment instances
+        var attachments = Set<SPTextAttachment>()
+
+        textView.attributedString().enumerateAttachments(of: SPTextAttachment.self) { (attachment, range) in
+            XCTAssertFalse(attachments.contains(attachment))
+            attachments.insert(attachment)
+        }
+
+        XCTAssertEqual(attachments.count, 2)
+    }
+}
+
+
+// MARK: - Helpers!
+//
+private extension NSTextViewSimplenoteTests {
+
+    var samplesForIndentation: [(text: String, indented: String)] {
+        let marker = String.attachmentString
+        return [
+            (text: "- L1",          indented: "\t- L1"),
+            (text: "\t- L2",        indented: "\t\t- L2"),
+            (text: "  - L3",        indented: "\t  - L3"),
+            (text: "* L1",          indented: "\t* L1"),
+            (text: "\t* L2",        indented: "\t\t* L2"),
+            (text: "  * L3",        indented: "\t  * L3"),
+            (text: "+ L1",          indented: "\t+ L1"),
+            (text: "\t+ L2",        indented: "\t\t+ L2"),
+            (text: "  + L3",        indented: "\t  + L3"),
+            (text: marker + " L1",  indented: "\t" + marker + " L1"),
+        ]
+    }
+
+    var samplesForNewline: [(text: String, enhanced: String)] {
+        let marker = String.attachmentString
+        return [
+            (text: "- L1",          enhanced: "- L1\n- "),
+            (text: "\t- L2",        enhanced: "\t- L2\n\t- "),
+            (text: "  - L3",        enhanced: "  - L3\n  - "),
+            (text: "* L1",          enhanced: "* L1\n* "),
+            (text: "\t* L2",        enhanced: "\t* L2\n\t* "),
+            (text: "  * L3",        enhanced: "  * L3\n  * "),
+            (text: "+ L1",          enhanced: "+ L1\n+ "),
+            (text: "\t+ L2",        enhanced: "\t+ L2\n\t+ "),
+            (text: "  + L3",        enhanced: "  + L3\n  + "),
+            (text: marker + " L1",  enhanced: marker + " L1\n" + marker + String.space),
+        ]
+    }
+
+    var samplesForNewlineRemoval: [String] {
+        let marker = String.attachmentString
+        return [
+            "-",
+            "\t- ",
+            "  - ",
+            "* ",
+            "\t* ",
+            "  * ",
+            "+ ",
+            "\t+ ",
+            "  + ",
+            marker,
+            "\t\t\t" + marker
+        ]
+    }
+
+    var samplePlainText: [String] {
+        return [
+            "Here comes the sun, here comes the sun\n",
```

Comment:
Maybe we shouldn't use copyrighted song lyrics here :)

Reply:
You know... I thought it was already public domain, but huge nope!! Updating, thanks for the heads up!!